include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -5483,10 +5484,13 @@ struct OrtApi {
    *
    * Supports initializers defined in an outer scope (i.e., a parent graph).
    *
+   * Supports initializers stored in an external file. For external initializers, ORT memory maps
+   * the initializer data on the first call to this function. If caller needs custom memory mapping,
+   * use ValueInfo_GetExternalInitializerInfo to get the location of the initializer data.
+   *
    * \param[in] value_info The OrtValueInfo instance.
-   * \param[out] initializer_value Output parameter set to the initializer value or NULL. The OrtValue data pointer
-   *                               (obtained via GetTensorData) is stable during the lifetime of the OrtSession
-   *                               that owns the OrtGraph.
+   * \param[out] initializer_value Output parameter set to the initializer value or NULL. Do not cache the OrtValue
+   *                               as it is released when the owning OrtGraph is released.
    *
```

Comment:
I thought we wanted EPs to cache it. OrtValues are shared Ptrs so why not to cache them if they needed?
We would need a way to clone it though, but I can see it come useful at some point.

Reply:
Because of this recent change to not cache the `OrtValue` (for external) in `Graph`, the memory mapped data will be released when the `OrtGraph` is released (after GetCapability and after Compile). That's another downside of not caching it in Graph.